### DISCUSSION TITLE
Specify SMTP credentials when sending

### DISCUSSION THREAD

**Discussion by willjasen** [postalserver/postal#2802] [Help with using Postal] [answered]
This is just completely getting past me but I can't seem to find it within the interface or the documentation - how do you specify credentials when sending? In my example, I want to submit outgoing emails to SendGrid and I have the hostname, port, and SSL mode, but where do you input the API key? I would think it would be under the SMTP Endpoints page but I can't find it. I'm probably completely just looking over it..

**Comment by willpower232** [comment, marked as answer]
Are you trying to relay all messages out of Postal to SendGrid? have a look at #48

**Reply by willjasen** [reply]
That was exactly what I needed, thanks!

To summarize for others, you'll want to:

1. copy [smtp_sender.rb](https://raw.githubusercontent.com/postalserver/postal/2e7aaaba5a9a37f32bdff9a9fde0379834b2a4db/lib/postal/smtp_sender.rb) to /opt/postal/smtp_sender.rb
2. edit line 66 of /opt/postal/smtp_sender.rb to reflect:
```
smtp_client.start(@source_ip_address ? @source_ip_address.hostname : self.class.default_helo_hostname, server[:user], server[:password])
```
3. edit line 261 of /opt/postal/smtp_sender.rb to reflect:

```
ssl_mode: relay.ssl_mode,
user: relay.user,
password: relay.password
```

4. edit /opt/postal/install/docker-compose.yml to include a new volume under the worker container:
```
- /opt/postal/smtp_sender.rb:/opt/postal/app/lib/postal/smtp_sender.rb
```

6. edit /opt/postal/config/postal.yml to include:
```
smtp_relays:
-
    hostname: smtp.sendgrid.net
    port: 587
    ssl_mode: Auto
    user: apikey
    password: YOUR-API-KEY
```
7. restart the postal worker container via `docker restart postal-worker-1` or restart all of postal via `postal restart`

Test sending a message out to verify it goes through!
